usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js

```diff
@@ -8,119 +8,348 @@
+                    console.log("EXCEEDED EXPIRATION OF TOKEN");
+                    Script.clearInterval(that.interval);
+                    that.interval = false;
+                    return;
+                }
+                that.request(that.scheduleURL, function (error, response) {
+                    if (error) {
+                        console.log("could not complete request", error);
+                        Script.clearInterval(that.interval);
+                        that.interval = false;
+                        return;
+                    } else {
+                        that.clearEventList(that.entityID);
+                        console.log(JSON.stringify(response));
+                        var events = response.items;
+                        if (events.length > 0) {
+                            for (var i = 0; i < events.length; i++) {
+                                var event = events[i];
+                                var start = event.start.dateTime;
+                                var end = event.end.dateTime;
+                                var summary = event.summary;
+                                if (!start) {
+                                    start = event.start.date;
+                                }        
+                                var startTimestamp = that.googleDateToUTCDate(start);
+                                var endTimestamp = that.googleDateToUTCDate(end);
+                  
+                                that.postEvents(that.entityID, [summary, startTimestamp, endTimestamp]);
+                            }
+                        } else {
+                            that.showNoScheduledEvents(that.entityID);
+                        }
+                    } 
+                });
+            }, INTERVAL_FREQUENCY_MS);
         }
     };
 
 
-    this.showNoScheduledEvents = function(id, params) {
+    var googleDate = /^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})([+-]\d{2}):(\d{2})$/;
+    var MINS_PER_HOUR = 60;
+    this.googleDateToUTCDate = function(d) {
+        var m = googleDate.exec(d);
+        var year = +m[1];
+        var month = +m[2];
+        var day = +m[3];
+        var hour = +m[4];
+        var minute = +m[5];
+        var second = +m[6];
+        var msec = 0;
+        var tzHour = +m[7];
+        var tzMin = +m[8];
+        var tzOffset = new Date().getTimezoneOffset() + tzHour * MINS_PER_HOUR + tzMin;
+        var date = new Date(Date.UTC(year, month - 1, day, hour, minute - tzOffset, second, msec));
+        return date;
+    };
+
+
+    this.showNoScheduledEvents = function(id) {
         if (id === that.entityID) {
-            Entities.editEntity(that.room.scheduleEntityID, {text: "No events scheduled for this room"});
+            Entities.editEntity(id, {text: "Nothing on the schedule for this room today."});
         }
     };
 
 
-    this.addEvent = function(id, params) {
+    this.postEvents = function(id, params) {
```

Comment:
why does this function take an array as its second argument instead of listing out the three arguments within that array? (`summary, startTimestamp, endTimestamp`)

Reply:
also, same question as above - why is this function not global? i'm asking myself this question a few times throughout the code